src/server/plugins/engine/components/NumberField.ts

```diff
@@ -82,8 +82,13 @@ export class NumberField extends FormComponent {
     const viewModel = super.getViewModel(payload, errors)
     let { attributes, prefix, suffix, value } = viewModel
 
-    if (schema.precision) {
-      attributes.step = '0.' + '1'.padStart(schema.precision, '0')
+    if (
+      !schema.precision ||
+      (NumberField.isNumber(schema.precision) && schema.precision <= 0)
+    ) {
```

Comment:
Can use `this.isValue()` versus `NumberField.isNumber()` when inside the class

But might be better reserving the **isValue** and **isState** helpers for payload/state only?

```suggestion
    if (typeof schema.precision === 'undefined' || schema.precision <= 0) {
```

Reply:
Resolved with 25530c8557fc1d29375036effe886994e9692944

Reply:
Makes sense if we hit this more in future, cheers